tests/app/config/test_config_validation.py

```diff
@@ -1,246 +1,210 @@
-    validator = StorageConfig(storage)
-    validator.validate()
-    assert validator.errors == []
-
-    # Test with wrong storage type
-    storage = {"type": "xyz"}
-    validator = StorageConfig(storage)
-    validator.validate()
-    assert validator.errors == [
-        "Error: Unsupported storage type 'xyz' specified in the configuration.Valid values are ['file', 'db']"
-    ]
-
-    # Test with storage type as `db` wrong `db` value
-    storage = {"type": "db", "db": "db123", "name": "pebblo_db"}
-    validator = StorageConfig(storage)
-    validator.validate()
-    assert validator.errors == [
-        "Error: Unsupported db type 'db123' specified in the configuration.Valid values are ['sqlite']"
-    ]
-
-    # Test with storage type as `db` without `db` and `name`
-    storage = {"type": "db"}
-    validator = StorageConfig(storage)
-    validator.validate()
-    assert validator.errors == [
-        "Error: Unsupported db type 'None' specified in the configuration.Valid values are ['sqlite']",
-        "Error: Unsupported db name 'None specified in the configurationString values are allowed only",
-    ]
-
-
-def test_validate_config(setup_and_teardown):
-    # Test with valid configuration
-    config = {
-        "daemon": {"host": "localhost", "port": "8000"},
-        "logging": {"level": "info"},
-        "reports": {
-            "format": "pdf",
-            "renderer": "xhtml2pdf",
-            "cacheDir": "~/.pebblo_test_",
-        },
-        "classifier": {
-            "mode": "all",
-            "anonymizeSnippets": True,
-        },
-        "storage": {"type": "file"},
-    }
-    validate_config(config)
-    # If the configuration is valid, validate_config should not raise any exceptions
-
-    # Test with invalid configuration
-    config = {
-        "daemon": {"host": 123, "port": "8000"},
-        "logging": {"level": "invalid_level"},
-        "reports": {
-            "format": "doc",
-            "renderer": "xhtml2pdf",
-            "cacheDir": "~/.pebblo_test_",
-        },
-        "classifier": {
-            "anonymizeSnippets": "abc",
-        },
-    }
-    with pytest.raises(SystemExit):
-        validate_config(config)
-    # If the configuration is invalid, validate_config should raise a SystemExit exception
+        "renderer": "xhtml2pdf",
+        "cacheDir": "~/.pebblo",
+        "anonymizeSnippets": True,
+    },
+    "classifier": {"mode": "all"},
+    "storage": {"type": "file"},
+}
+
+
+def test_with_valid_values():
+    valid_data = Config.parse_obj(config_json)
+    assert valid_data == Config(**config_json)
+
+
+def test_daemon_config_validate_invalid_port():
+    config_json.update({"daemon": {"host": "localhost", "port": "12345678"}})
+    with pytest.raises(Exception) as err_msg:
+        Config.parse_obj(config_json)
+    error_msg = """daemon.port
+  Value error, Error: Invalid port '12345678'. Port must be between 1 and 65535."""
+    assert error_msg in str(err_msg.value)
```

Comment:
One UT with invalid port value as a string like `invalid_string` is worth adding here.